bot/commands/misc/convert.js

```diff
@@ -19,30 +19,34 @@ async function convertCurrency (msg, baseValue, baseType, targetType, prefix) {
 		}
 		const conversions = await res.json();
 		const result = conversions.rates[targetType] * baseValue;
-		msg.channel.createMessage(`${baseValue}${baseType} is roughly equal to ${result.toFixed(2)}${targetType}`);
+		return `${baseValue}${baseType} is roughly equal to ${result.toFixed(2)}${targetType}`;
 	} catch (err) {
-		msg.channel.createMessage(err.message).catch(() => {});
+		return err.message;
 	}
 }
 
-module.exports = new Command('convert', (msg, args, context) => {
+module.exports = new Command('convert', async (msg, args, context) => {
 	if (args.length === 0) {
 		msg.channel.createMessage(`For currency codes visit: <https://www.xe.com/en/iso4217.php>
-		\rFor units visit: <https://www.npmjs.com/package/convert-units>
-		\rCommand can be used with: \`${context.prefix}convert [amount] [baseUnit] [targetUnit]\``);
+For units visit: <https://www.npmjs.com/package/convert-units>
+Command can be used with: \`${context.prefix}convert [amount] [baseUnit] [targetUnit]\``);
 		return;
 	}
+	const baseValue = parseFloat(args[0].replace(/,/g, '.')); // replace commas with dots since some people use those for decimals
 
-	const baseValue = args[0].replace(/,/g, '.'); // replace commas with dots since some people use those for decimals
+	if (isNaN(baseValue)) {
+		msg.channel.createMessage('Please enter a valid number!');
+		return;
+	}
 	const baseType = args[1];
 	const targetType = args[2];
 	const prefix = context.prefix;
 	try {
 		// see if it can convert it normally first
-		convertUnits(msg, baseValue, baseType, targetType);
+		msg.channel.createMessage(convertUnits(baseValue, baseType, targetType));
```

Comment:
Possible unhandled promise rejection. Assuming we want to do nothing if the bot doesn't have message permissions in the channel:

```suggestion
		msg.channel.createMessage(convertUnits(baseValue, baseType, targetType)).catch(() => {});
```